app/models/project_role.rb

```diff
@@ -91,8 +91,14 @@ def validate_other_selections
   end
 
   def validate_one_primary_pi
-    if self.protocol.project_roles.where(role: "primary-pi").include?(self) && self.role != "primary-pi"
-      errors.add(:role, "- Protocols must have a Primary PI.")
+
+    primary_pi = protocol.project_roles.where(role: 'primary-pi')
+
+    if primary_pi.empty?
+      errors.add(:must, "include one Primary PI.")
+      return false
+    elsif self.role == 'primary-pi' && !primary_pi.include?(self) && primary_pi.size > 0
+      errors.add(:role, "- This protocol already has a Primary PI.")
```

Comment:
Your changes to #validate_one_primary_pi allows users to remove Primary PI's from protocols when editing their associated users (protocols must have one Primary PI). This may be what you're looking for, fixing the project_role_spec:

```
def validate_one_primary_pi
    pi_roles = self.protocol.project_roles.where(role: "primary-pi")
    if pi_roles.empty? || pi_roles.include?(self) && self.role != "primary-pi"
      errors.add(:role, "- Protocols must have a Primary PI.")
      false
    else
      true
    end
  end
```